### ISSUE TITLE
Add timeout annotations for resources

### ISSUE THREAD

**Issue by kriswuollett** [GoogleContainerTools/kpt#3874]
Not all operators follow the same conventions, see https://github.com/keycloak/keycloak/issues/17515. If we know for a fact that a CRD may not get detected as reconciled, then it would be great to specify timeouts at the resource level than at the `live apply` command invocation level.

There are likely other use cases as well, like allowing 30 minute timeouts for specific resources like an `Ingress` resource if a vendor-provided load balancer needs to be created, but specifying a 30 minute timeout on the command line for all resources is inappropriate.

**Comment by mortent** [comment]
@kriswuollett Yeah, this is something we have discussed in the past, but it has never ended up being a priority. It probably requires changes in the underlying cli-utils library, but we are open to doing that. 

This isn't a top priority, but we are more than happy to provide help and guidance if you want to add this feature. Just remember that the dependency graph is flattened into phases before the apply actually happens, so that might limit the usefulness somewhat. But we have a long-term goal of implementing an asynchronous dependency graph as mentioned in https://github.com/GoogleContainerTools/kpt/issues/3395#issuecomment-1191138981.

**Comment by kriswuollett** [comment]
@mortent ok, it is good to know that the feature has been discussed and open to it being added. At the moment I'm just trying to get a variety of things pieced together, and since this affects only one `ResourceGroup`, I can live with command line timeout for now. If I get some time to dig into it in the future, I'll swing back and look into implementing it.
